test/mpi.py

```diff
@@ -23,36 +23,251 @@
+import string
 
+from vsc.utils.run import run_simple
 from vsc.mympirun.factory import getinstance
-from vsc.mympirun.mpi.mpi import MPI
+import vsc.mympirun.mpi.mpi as mpim
+#from vsc.mympirun.mpi.openmpi import OpenMPI
 from vsc.mympirun.rm.local import Local
 from vsc.mympirun.option import MympirunOption
 
 # we wish to use the mpirun we ship
 os.environ["PATH"] = os.path.dirname(os.path.realpath(__file__)) + os.pathsep + os.environ["PATH"]
 
 
-class TestMPI(TestCase):
-    """Test for the mpi class."""
+class TestMPI(unittest.TestCase):
 
-    def test_MPI_local(self):
-        """"Test the MPI class with the local scheduler"""
-        # options
-        # m = MympirunOption()
-        # m.args = ['echo', 'foo']
-        # # should not throw an error
-        # mpi_instance = getinstance(MPI, Local, m)
-        # mpi_instance.main()
+    """tests for vsc.mympirun.mpi.mpi functions"""
 
-        # # check for correct .mpd.conf file
-        # mpdconffn = os.path.expanduser('~/.mpd.conf')
-        # perms = stat.S_IMODE(os.stat(mpdconffn).st_mode)
-        # self.assertEqual(perms, 0400, msg='permissions %0o for mpd.conf %s' % (perms, mpdconffn))
+    #######################
+    ## General functions ##
+    #######################
+
+    def test_what_mpi(self):
+        """test if what_mpi returns the correct mpi flavor"""
+        scriptnames = ["ompirun", "mpirun", "impirun", "mympirun"]
+        for scriptname in scriptnames:
+            # if the scriptname is an executable located on this machine
+            if mpim.which(scriptname):
+                (returned_scriptname, mpi, found) = mpim.what_mpi(scriptname)
+                print("what mpi returns: %s, %s, %s" % (returned_scriptname, mpi, found))
+                # if an mpi implementation was found
+                if mpi:
+                    self.assertTrue(mpi in found,
+                                    msg="returned mpi (%s) is not an element of found_mpi (%s)" % (mpi, found))
+                    self.assertTrue(returned_scriptname == scriptname,
+                                    msg="returned scriptname (%s) doesn't match actual scriptname (%s)" %
+                                    (returned_scriptname, scriptname))
+                else:
+                    self.assertTrue(returned_scriptname.endswith("mpirun"),
+                                    msg="no mpi found, scriptname should be the path to mpirun but is %s" %
+                                    returned_scriptname)
+
+    def test_stripfake(self):
+        """Test if stripfake actually removes the /bin/fake path in $PATH"""
+        print("old path: %s" % os.environ["PATH"])
+        mpim.stripfake()
+        newpath = os.environ["PATH"]
+        self.assertFalse(("bin/%s" % mpim.FAKE_SUBDIRECTORY_NAME) in newpath, msg="the faked dir is still in $PATH")
+
+    def test_which(self):
+        """test if which returns a path that corresponds to unix which"""
+
+        testnames = ["python", "java"]
```

Comment:
`pwd` is a binary? Anyway, take your pick. As long as it's something more commonly available then java :wink: